legend-engine-xt-relationalStore-executionPlan-authorizer/src/main/java/org/finos/legend/engine/plan/execution/authorization/RelationalMiddleTierPlanExecutionAuthorizer.java

```diff
@@ -215,15 +217,47 @@ private ExecutionAuthorization evaluateCredentialAuthorization(Identity identity
         }
     }
 
+    private void collectSQLExecutionNodes(ExecutionNode executionNode, MutableList<SQLExecutionNode> sqlExecutionNodes )
+    {
+        if (executionNode == null)
+        {
+            return;
+        }
+        if (executionNode instanceof SQLExecutionNode)
+        {
+            sqlExecutionNodes.add((SQLExecutionNode)executionNode);
+            return;
+        }
+        if (executionNode instanceof GlobalGraphFetchExecutionNode)
+        {
+            GlobalGraphFetchExecutionNode ggfe = (GlobalGraphFetchExecutionNode) executionNode;
+            if (ggfe.children != null)
+            {
+                ListIterate.forEach(ggfe.children, c -> this.collectSQLExecutionNodes(c, sqlExecutionNodes));
+            }
+            this.collectSQLExecutionNodes(ggfe.localGraphFetchExecutionNode, sqlExecutionNodes);
+        }
+        else if (executionNode instanceof LocalGraphFetchExecutionNode)
```

Comment:
Is this branch necessary? Won't the final if statement (on line 248) do the same as this if block?